packages/feature-flags/src/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @typedef {Function} Logger
+ * @param {...string} args
+ * @returns {void}
+ * */
+
+/**
+ * @param {{ info: Logger, error: Logger }} logger
+ * @param {string} [endpoint]
+ * @returns {(featureFlagName: string, localPlanningAuthorityCode?: string) => Promise<boolean>}
+ * */
+export const isFeatureActive =
+	(logger, endpoint) => async (featureFlagName, localPlanningAuthorityCode) => {
+		if (process.env.FEATURE_FLAGS_SETTING === 'ALL_ON') {
+			return true;
+		}
+
+		if (endpoint === undefined) {
+			logger.info('Returning false for all feature flags because no endpoint provided.');
+			return false;
+		}
+
+		if (endpoint.length <= 1) {
+			logger.error(`Endpoint value ${endpoint} is invalid.`);
+			return false;
+		}
+
+		const { AppConfigurationClient } = require('@azure/app-configuration');
+		const appConfigClient = new AppConfigurationClient(endpoint);
```

Comment:
Is it worth doing this at the first function level, and renaming?

Then you can create a shared client for use within the app, e.g.

```
const isFeatureActive = newFeaturesClient(logger, endpoint);

// ... elsewhere
isFeatureActive ('flag-1');
isFeatureActive ('flag-2');
```